service/history/task/transfer_active_task_executor_test.go

```diff
@@ -706,6 +707,89 @@ func (s *transferActiveTaskExecutorSuite) TestProcessCloseExecution_HasParent()
+		RunID:      uuid.New(),
+	}
+	s.mockDomainCache.EXPECT().GetDomainByID(parentDomainID).Return(s.remoteTargetDomainEntry, nil).AnyTimes()
+
+	mutableState := execution.NewMutableStateBuilderWithVersionHistoriesWithEventV2(
+		s.mockShard,
+		s.logger,
+		s.version,
+		workflowExecution.GetRunID(),
+		s.domainEntry,
+	)
+	_, err := mutableState.AddWorkflowExecutionStartedEvent(
+		workflowExecution,
+		&types.HistoryStartWorkflowExecutionRequest{
+			DomainUUID: s.domainID,
+			StartRequest: &types.StartWorkflowExecutionRequest{
+				WorkflowType:                        &types.WorkflowType{Name: workflowType},
+				TaskList:                            &types.TaskList{Name: taskListName},
+				ExecutionStartToCloseTimeoutSeconds: common.Int32Ptr(2),
+				TaskStartToCloseTimeoutSeconds:      common.Int32Ptr(1),
+			},
+			ParentExecutionInfo: &types.ParentExecutionInfo{
+				DomainUUID:  parentDomainID,
+				Domain:      parentDomainName,
+				Execution:   &parentExecution,
+				InitiatedID: parentInitiatedID,
+			},
+		},
+	)
+	s.Nil(err)
+
+	di := test.AddDecisionTaskScheduledEvent(mutableState)
+	event := test.AddDecisionTaskStartedEvent(mutableState, di.ScheduleID, taskListName, uuid.New())
+	di.StartedID = event.GetEventID()
+	event = test.AddDecisionTaskCompletedEvent(mutableState, di.ScheduleID, di.StartedID, nil, "some random identity")
+
+	taskID := int64(59)
+	event = test.AddCompleteWorkflowEvent(mutableState, event.GetEventID(), nil)
+
+	transferTask := s.newTransferTaskFromInfo(&persistence.TransferTaskInfo{
+		Version:    s.version,
+		DomainID:   s.domainID,
+		WorkflowID: workflowExecution.GetWorkflowID(),
+		RunID:      workflowExecution.GetRunID(),
+		TaskID:     taskID,
+		TaskList:   taskListName,
+		TaskType:   persistence.TransferTaskTypeCloseExecution,
+		ScheduleID: event.GetEventID(),
+	})
+
+	persistenceMutableState := s.createPersistenceMutableState(mutableState, event.GetEventID(), event.GetVersion())
+	s.mockExecutionMgr.On("GetWorkflowExecution", mock.Anything, mock.Anything).Return(&persistence.GetWorkflowExecutionResponse{State: persistenceMutableState}, nil)
+	s.mockHistoryClient.EXPECT().RecordChildExecutionCompleted(gomock.Any(), &types.RecordChildExecutionCompletedRequest{
+		DomainUUID:         parentDomainID,
+		WorkflowExecution:  &parentExecution,
+		InitiatedID:        parentInitiatedID,
+		CompletedExecution: &workflowExecution,
+		CompletionEvent:    event,
+	}).Return(nil).AnyTimes()
+	s.mockVisibilityMgr.On("RecordWorkflowExecutionClosed", mock.Anything, mock.Anything).Return(nil).Once()
+	s.mockArchivalMetadata.On("GetVisibilityConfig").Return(archiver.NewDisabledArchvialConfig())
+	s.mockExecutionMgr.On("UpdateWorkflowExecution", mock.Anything, mock.Anything).Return(
+		&persistence.UpdateWorkflowExecutionResponse{MutableStateUpdateSessionStats: nil}, nil)
+	s.mockClusterMetadata.EXPECT().ClusterNameForFailoverVersion(int64(-24)).Return(s.mockClusterMetadata.GetCurrentClusterName()).AnyTimes()
```

Comment:
Not sure where -24 is coming from, any idea? This is s.version until we close the execution